src/shapes/Text/StyledText.ts

```diff
@@ -181,18 +181,28 @@ export abstract class StyledText<
     }
   }
 
-  private _extendStyles(index: number, styles: TextStyleDeclaration): void {
+  private _extendStyles(index: number, style: TextStyleDeclaration): void {
     const { lineIndex, charIndex } = this.get2DCursorLocation(index);
 
     if (!this._getLineStyle(lineIndex)) {
       this._setLineStyle(lineIndex);
     }
 
-    if (!Object.keys(this._getStyleDeclaration(lineIndex, charIndex)).length) {
-      this._setStyleDeclaration(lineIndex, charIndex, {});
-    }
+    // first create a new object that is a merge of existing and new
+    const newStyle: TextStyleDeclaration = {
+      ...this._getStyleDeclaration(lineIndex, charIndex),
+      ...style,
+    };
+
+    // then delete what is undefined in styles from newStyle
+    Object.keys(style).forEach((key) => {
+      if (style[key as keyof TextStyleDeclaration] === undefined) {
+        delete newStyle[key as keyof TextStyleDeclaration];
+      }
+    });
```

Comment:
so newStyle is created correctly right now, you don't like the shape of the code that creates it.

if i use filter, i have to:
- mix the 2 object together, creating a new one 
- use Entries, that is a loop + creation of an array
- filter that returns the new list of entries we want to keep
- fromEntries that create a new object from the filtered list

this is more compact, but i can't call it more straight  forward at all:
```js
 Object.fromEntries(Object.entries({
    ...this._getStyleDeclaration(lineIndex, charIndex),
    ...style,
  }).filter(([key, value]) => value !== undefined));
```

I can change it but you didn't mention what is unpreferable about the actual code.

The original comment was:

```
Since you are creating a new object I think it is better to filter out undefined beforehand instead of creating it and then deleting stuff
```

We are still creating a new object with all the stuff ( the spread statement ), then we are creating an array of arrays with all entries, a new one filtered, then we are creating a new object once again.

Is that you don't want to carry over an object on which has been used the delete keyword?

Reply:
With pick is overly complicated

Reply:
pickby gives us a negative code size and probably the same speed, while the fromEntries <-> toEntries is definetely worse on both aspects

Reply:
pickBy,yes

Reply:
nice!